spec/requests/openid_connect_token_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe 'OpenID Connect Token endpoint' do
+  describe 'token endpoint' do
+    context 'origin is included in ServiceProvider redirect_uris' do
+      it 'does not include IDP session cookie' do
+        user = create(:user)
+        client_id = 'urn:gov:gsa:openidconnect:test'
+        service_provider = build(:service_provider, issuer: client_id)
+        identity = IdentityLinker.new(user, service_provider).link_identity(
+          rails_session_id: SecureRandom.hex,
+          ial: 1,
+        )
+
+        jwt_payload = {
+          iss: client_id,
+          sub: client_id,
+          aud: api_openid_connect_token_url,
+          jti: SecureRandom.hex,
+          exp: 5.minutes.from_now.to_i,
+        }
+
+        client_private_key = OpenSSL::PKey::RSA.new(
+          Rails.root.join(
+            'keys',
+            'saml_test_sp.key',
+          ).read,
+        )
+
+        client_assertion = JWT.encode(jwt_payload, client_private_key, 'RS256')
+
+        params = {
+          grant_type: 'authorization_code',
+          code: identity.session_uuid,
+          client_assertion_type: OpenidConnectTokenForm::CLIENT_ASSERTION_TYPE,
+          client_assertion: client_assertion,
+        }
+        post api_openid_connect_token_path, params: params, headers: { 'HTTP_ORIGIN' => 'https://example.com' }
```

Comment:
is there a simpler endpoint that would also highlight the same behavior? like maybe the openid certs controller?

Reply:
It unfortunately needs to try to access or mutate the session to show the behavior (even if it's only via analytics). Security events, OIDC userinfo, and Attempts API are the other options that reproduce the issue and it looks like userinfo is probably the simplest, so I replaced it in aa6baf12968814988a73fa3b1a0f04df44565230